protocol.go

```diff
@@ -109,9 +116,28 @@ func ProtocolNameToID(name string) ProtocolID {
 // All registered protocols will be copied to every instantiated Server. If a
 // protocol is tied to a service, use `Server.ProtocolRegisterName`
 func GlobalProtocolRegister(name string, protocol NewProtocol) (ProtocolID, error) {
+	protocols.Lock()
+	defer protocols.Unlock()
+	if protocols.serverStarted {
+		log.Lvl4("Cannot call 'GlobalProtocolRegister' when a server has already started.")
```

Comment:
Shouldn't this do `return nil, errors.New()`?

Reply:
In fact I wanted to put `log.Fatal` at first (I don't know why I ended up putting `log.Lvl4`), to fail directly. The main reason was because this function is currently used a lot in Cothority (and Onet), and it (almost) never checks the returned error, so returning the error would currently not do anything. Also, one of @jeffallen comment on the issue was: "Goal: give early, clear crash/error so that students don't fall into this trap.".

I could also return the error as you suggest, but then, before that, I'll have to go over all the calls to this function in Cothority and Onet and add the code to check the returned error, and fail/return the error further.

What do you think makes more sense in our case?

Reply:
that's horrible! I feel very bad ;) Then do a `panic` here, so you can catch it in the tests and look if it behaves as needed: https://blog.golang.org/defer-panic-and-recover